test/suites/greenboot.robot

```diff
@@ -0,0 +1,144 @@
+${GREENBOOT_CONFIG_CONTENT}     MICROSHIFT_WAIT_TIMEOUT_SEC=180
+${MANIFEST_SOURCE_DIR}          ./assets/kustomizations/greenboot/
+${MANIFEST_DIR}                 /etc/microshift/manifests
+${HOSTNAME_BIN_PATH}            ""
+
+
+*** Test Cases ***
+GreenbootHealth Run with User Workload
+    [Documentation]    Add user's workload and verify that the workload starts and greenboot is successful.
+    Greenboot Healthy
+    Add User Workload
+    Microshift Cleanup And Start
+
+    [Teardown]    Run Keywords
+    ...    Cleanup User Workload
+    ...    Microshift Cleanup And Start
+
+GreenbootHealth Microshift Service Failure
+    [Documentation]    Microshift Service failure
+    Greenboot Healthy
+    Microshift Disrupt Service
+    Cleanup MicroShift    "--keep-images"
+    Run Keyword And Expect Error    0 != 1
+    ...    Systemctl    enable    --now microshift
+    Run Keyword And Expect Error    0 != 1
+    ...    Greenboot Healthy
+
+    [Teardown]    Run Keywords
+    ...    Microshift Restore Service
+    ...    Microshift Cleanup And Start
+
+GreenbootHealth Microshift Pod Failure
+    [Documentation]    Microshift pod network failure
+    Greenboot Healthy
+    Microshift Disrupt Pod Network
+    Restart MicroShift
+    Run Keyword And Expect Error    0 != 1
+    ...    Greenboot Healthy
+
+    [Teardown]    Run Keywords
+    ...    Restore Default MicroShift Config
+    ...    Microshift Cleanup And Start
+
+
+*** Keywords ***
+Setup Suite
+    [Documentation]    Set up all of the tests in this suite
+    Check Required Env Variables
+    Login MicroShift Host
+    Setup Kubeconfig
+    Upload String To File    ${GREENBOOT_CONFIG_CONTENT}    ${GREENBOOT_CONFIG_FILE} # change the default timeout
+
+Add User Workload
+    [Documentation]    Upload User workload files to the MicroShift host
+    Put File    ${LOCAL_WORKLOAD_FILE}    /tmp/busybox_running_check.sh
+    ${stdout}    ${rc}=    Execute Command
+    ...    [ -f ${GREENBOOT_WORKLOAD_FILE} ] || sudo mv /tmp/busybox_running_check.sh ${GREENBOOT_WORKLOAD_FILE}
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+    Put Directory    ${MANIFEST_SOURCE_DIR}    /tmp/manifests
+    ${stdout}    ${rc}=    Execute Command
+    ...    cp -f /tmp/manifests/* ${MANIFEST_DIR} && sudo rm -rf /tmp/manifests
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+Cleanup User Workload
+    [Documentation]    Remove User workload files from the MicroShift host
+    ${stdout}    ${rc}=    Execute Command
+    ...    rm -rf ${GREENBOOT_WORKLOAD_FILE} ${MANIFEST_DIR}/*
+    ...    sudo=True    return_rc=True
+    Should Be Equal As Integers    0    ${rc}
+
+Greenboot Healthy
+    [Documentation]    Restart the greenboot-healthcheck service and check its status
+    Systemctl    restart    greenboot-healthcheck.service
+    Wait Until Greenboot Health Check Exited
+
+Microshift Disrupt Service
```

Comment:
I'd put verb first, so it reads naturally in the test case
```suggestion
Disrupt MicroShift
```
Same goes for other keywords starting with `MicroShift`

Reply:
i'll replace it with verb `Simulate`